internal/service/alarm_notification_searcher.go

```diff
@@ -0,0 +1,225 @@
+	b.pathIndexMap = &map[string]alarmSubIdSet{}
+	b.noFilterSubsSet = &alarmSubIdSet{}
+
+	// Create the filter expression parser:
+	selectorParser, err := search.NewSelectorParser().
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		b.logger.Error(
+			"failed to create filter expression parser: ",
+			slog.String("error: ", err.Error()),
+		)
+		return
+	}
+	b.selectorParser = selectorParser
+}
+
+// NOTE the function should be called by a function that is holding the semophone
+func (b *alarmSubscriptionSearcher) getSubFilters(filterStr string, subId string) (err error) {
+
+	//no filter found, return empty array and behavior as "*"
+	if filterStr == "" {
+		(*b.noFilterSubsSet)[subId] = struct{}{}
+		return
+	}
+
+	result, err := b.selectorParser.Parse(filterStr)
+
+	if err != nil {
+		return
+	}
+
+	subInfo := (*b.subscriptionInfoMap)[subId]
+	subInfo.filters = *result
+	(*b.subscriptionInfoMap)[subId] = subInfo
+
+	//for now use path 0 only
+	//to be fixed with full path for quicker search
+	for _, element := range result.Terms {
+		_, ok := (*b.pathIndexMap)[element.Path[0]]
+
+		if !ok {
+			(*b.pathIndexMap)[element.Path[0]] = alarmSubIdSet{}
+		}
+		(*b.pathIndexMap)[element.Path[0]][subId] = struct{}{}
+	}
+
+	return
+}
+
+// NOTE the function should be called by a function that is holding the semophone
+func (b *alarmSubscriptionSearcher) pocessSubscriptionMapForSearcher(subscriptionMap *map[string]data.Object,
+	jqTool *jq.Tool) (err error) {
+
+	for key, value := range *subscriptionMap {
+
+		(*b.subscriptionInfoMap)[key] = subscriptionInfo{}
+
+		subInfo := (*b.subscriptionInfoMap)[key]
+		//get uris
+		var uris string
+		err = jqTool.Evaluate(`.callback`, value, &uris)
+		if err != nil {
+			panic("the callback is mandary for subscription")
+		}
+		subInfo.uris = uris
+
+		var consumerId string
+		err = jqTool.Evaluate(`.consumerSubscriptionId`, value, &consumerId)
+		if err == nil {
+			subInfo.consumerSubscriptionId = consumerId
+		}
+
+		(*b.subscriptionInfoMap)[key] = subInfo
+
+		//get filter from data object
+		var filter string
+		err = jqTool.Evaluate(`.filter`, value, &filter)
+		if err != nil {
+			b.logger.Debug(
+				"Subscription  ", key,
+				" does not have filter included",
+			)
+		}
+
+		err = b.getSubFilters(filter, key)
+		if err != nil {
+			b.logger.Debug(
+				"pocessSubscriptionMapForSearcher ",
+				"subscription: ", key,
+				" error: ", err.Error(),
+			)
+		}
+	}
+	return
+}
+
+func (h *alarmNotificationHandler) getSubscriptionIdsFromAlarm(ctx context.Context, alarm data.Object) (result alarmSubIdSet) {
+
+	h.subscriptionMapMemoryLock.RLock()
+	defer h.subscriptionMapMemoryLock.RUnlock()
+	result = *h.subscriptionSearcher.noFilterSubsSet
+
+	subCheckedSet := alarmSubIdSet{}
+
+	for path, subSet := range *h.subscriptionSearcher.pathIndexMap {
+
+		var alarmPath string
+		path = fmt.Sprintf(`.%s`, path)
+		err := h.jqTool.Evaluate(path, alarm, &alarmPath)
+
+		if err == nil {
```

Comment:
Looks like if there is an error it is silently ignored. I'd suggest to first check if there is an error, handle it (maybe just log it) and then continue with the loop.

Reply:
Add the comment in the code. This is not error just not applied to the path